src/base/dom/pan-and-zoom.ts

```diff
@@ -84,6 +99,38 @@ export class PanAndZoom {
             startDistance = null;
             startPosition = null;
         });
+
+        let dragStartPosition: Vec2 | null = null;
+        let dragging = false;
+
+        this.target.addEventListener("mousedown", (e: MouseEvent) => {
+            if (e.button === 1 || e.button === 2) {
+                e.preventDefault();
+                dragging = true;
+                dragStartPosition = new Vec2(e.clientX, e.clientY);
+            }
+        });
+
+        this.target.addEventListener("mousemove", (e: MouseEvent) => {
+            if (dragging && dragStartPosition !== null) {
+                const currentPosition = new Vec2(e.clientX, e.clientY);
+                const delta = currentPosition.sub(dragStartPosition);
+                this.#handle_pan(-delta.x, -delta.y);
+                dragStartPosition = currentPosition;
+            }
+        });
+
+        this.target.addEventListener("mouseup", (e: MouseEvent) => {
+            if (e.button === 1 || e.button === 2) {
+                dragging = false;
+                dragStartPosition = null;
+            }
+        });
+
+        // コンテキストメニューの表示を防ぐためのイベントハンドラを追加
```

Comment:
Please keep code comments in English. Perhaps:

> Prevent the browser's default context menu.

Reply:
There were some comments left in Japanese. Since the suggested English is correct, I have replaced them.